### PR TITLE
add `Oscar.versioninfo()` function and short technical description of oscar architecture

### PR DESCRIPTION
**Pull request by benlorenz**
Full output might be also useful for an issue template:
```julia
julia> Oscar.versioninfo(jll=true,julia=true,branch=true)
OSCAR version 0.8.2-DEV - branch #bl/versioninfo
  combining:
    AbstractAlgebra.jl   v0.24.1
    GAP.jl               v0.7.7
    Hecke.jl             v0.11.0
    Nemo.jl              v0.29.2
    Polymake.jl          v0.7.1 - branch #bl/platform_error
    Singular.jl          v0.9.2 - branch #master
  building on:
    Antic_jll               v0.200.501+0
    Arb_jll                 v200.2200.0+0
    Calcium_jll             v0.400.102+0
    FLINT_jll               v200.800.401+1
    GAP_jll                 v400.1191.1+1
    Singular_jll            v403.0.100+0
    libpolymake_julia_jll   v0.8.0+0
    libsingular_julia_jll   v0.21.0+0
    msolve_jll              v0.1.7+0
    polymake_jll            v400.600.0+0
See `]st -m` for a full list of dependencies.

Julia Version 1.7.2
Commit bf53498635 (2022-02-06 15:21 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, tigerlake)
Environment:
  JULIA_CXX_RTTI = 1
  JULIA_CPU_TARGET = native
```

Architecture description should appear [here](https://oscar-system.github.io/Oscar.jl/previews/PR1122/General/architecture/) soon. 
cc: @lkastner  @OguzhanYueruek 

PS: Please complain if there are any further packages that should appear in these shortened dependency-lists.

### CODE REVIEW

**Review by lkastner — APPROVED** [review]
:heart:

File: docs/src/General/architecture.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+and interfaces provided by the julia packages for the cornerstones:
+  - ANTIC: [AbstractAlgebra.jl](https://github.com/Nemocas/AbstractAlgebra.jl),
+           [Hecke.jl](https://github.com/thofma/Hecke.jl),
+           [Nemo.jl](https://github.com/nemocas/Nemo.jl)
+  - [GAP.jl](https://github.com/oscar-system/GAP.jl)
+  - [Polymake.jl](https://github.com/oscar-system/Polymake.jl)
+  - [Singular.jl](https://github.com/oscar-system/Singular.jl)
+
+The packages are integrated into the julia package manager and will be
+installed automatically together with Oscar and can also be accessed directly
```
Comment (lkastner) [inline docs/src/General/architecture.md]:
```suggestion
installed automatically as dependencies of Oscar. They can be accessed directly
```
What do you think about this?

**Comment by thofma** [comment]
Very nice! Another option for the name would just be `Oscar.versioninfo()`, since we already have a namespace for Oscar things.

**Comment by benlorenz** [comment]
> Very nice! Another option for the name would just be `Oscar.versioninfo()`, since we already have a namespace for Oscar things.

Using `Oscar.versioninfo()` was actually what I implemented first but then I thought it might be better to have an `export`ed name and changed it to `oscar_versioninfo()`. I do slightly prefer the `Oscar.versioninfo` variant so I changed it back for now, but I am open for other suggestions.

**Review by OguzhanYueruek — APPROVED** [review]

**Comment by thofma** [comment]
Will merge tomorrow if there are no objections.

**Review by fingolfin — APPROVED** [review]
I like the idea

However, shouldn't `architecture.md` rather be in developer section? (We can easily change this later, of course)

**Comment by thofma** [comment]
@fingolfin What does that approve mean in connection with your comment? Should we merge this or not?

**Comment by lkastner** [comment]
This architecture section is also very useful for showing the users what parts make up Oscar.jl, which for me would push it more towards the general section. Anyway, as @fingolfin said, we can always move it around. I will merge this now, as I want to reference this in my developer introduction PR.

**Pull request merged**